### ISSUE TITLE
AlertDialog on iOS doesn't respect defined Theme

### ISSUE THREAD

**Issue by Cuyer** [JetBrains/compose-multiplatform#4524]
**Describe the bug**
AlertDialog on IOS doesn't respect application theme. AlertDialog on IOS doesn't respect AnimatedVisibility.

**Affected platforms**
- iOS

**Versions**
- Kotlin version*: 1.9.22
- Compose Multiplatform version*: 1.6.0-beta02
- OS version(s)* (required for Desktop and iOS issues): iOS 17.4
- OS architecture (x86 or arm64): arm64
- Device (model or simulator for iOS issues): Iphone 15

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
1. Create Theme.kt in commonMain
2. Apply colorScheme and Typography
3. Create AlertDialog and show it on the screen

**Expected behavior**
Dialog on iOS respects theme created in Theme.kt file.

**Screenshots**
Android:
![Screenshot_20240321_230042](https://github.com/JetBrains/compose-multiplatform/assets/51329471/d1157ff7-a4ce-4c6c-a5a3-8ea0ce21b7b4)

iOS:
![image](https://github.com/JetBrains/compose-multiplatform/assets/51329471/a601f514-14ec-4ee6-9411-d01c2433c89e)

**Additional context**
What's interesting is that if I add simple Text composable, then this theme is being applied.

Here is how I declare Theme:

```kotlin
@Composable
fun AppTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable() () -> Unit
) {
    val colorScheme = when {
        darkTheme -> darkScheme
        else -> lightScheme
    }
    CompositionLocalProvider(
        LocalSpacing provides Spacing()
    ) {
        MaterialTheme(
            colorScheme = colorScheme,
            typography = appTypography(),
            content = content,
            shapes = shapes
        )
    }
}
```

**Composable screen:**
```kotlin
class OnboardingScreen : Screen {
    @Composable
    override fun Content() {
        val onboardingModel = getScreenModel<OnboardingScreenModel>()
        val state = onboardingModel.state.collectAsState()
        val location = rememberLocation(noLocationCallback = { }).value

        var openDialog by remember {
            mutableStateOf(true)
        }

        AppDialog(
            confirmText = "Confirm",
            dialogText = "Sample text",
            dialogTitle = "Attention",
            dismissText = "Cancel",
            icon = Icons.Filled.Info,
            onConfirmation = { openDialog = !openDialog },
            onDismissRequest = { openDialog = !openDialog },
            openDialog = openDialog
        )
    }
}



Composable
fun AppDialog(
    modifier: Modifier = Modifier,
    onDismissRequest: () -> Unit,
    onConfirmation: () -> Unit,
    dialogTitle: String,
    dialogText: String,
    icon: ImageVector,
    confirmText: String,
    dismissText: String,
    openDialog: Boolean
) {
    AnimatedVisibility(
        visible = openDialog,
        enter = fadeIn(tween()),
        exit = fadeOut(tween())
    ) {
        AlertDialog(
            modifier = modifier,
            icon = { Icon(icon, contentDescription = "Content description") },
            title = { Text(dialogTitle) },
            text = { Text(dialogText) },
            confirmButton = {
                TextButton(
                    onClick = {
                        onConfirmation()
                    }
                ) {
                    Text(confirmText)
                }
            },
            dismissButton = {
                TextButton(
                    onClick = {
                        onDismissRequest()
                    }
                ) {
                    Text(dismissText)
                }
            },
            onDismissRequest = { onDismissRequest() },
        )
    }

}
```

**Comment by alexzhirkevich** [comment]
It was already fixed in a stable 1.6

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4524">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
